applications/MeshingApplication/custom_processes/internal_variables_interpolation_process.h

```diff
@@ -0,0 +1,1015 @@
+    double mWeight;                             // The integration weight of the GP
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+
+    ///@}
+    ///@name Private  Access
+    ///@{
+    ///@}
+
+    ///@}
+    ///@name Serialization
+    ///@{
+
+    ///@name Private Inquiry
+    ///@{
+    ///@}
+
+    ///@name Unaccessible methods
+    ///@{
+    ///@}
+}; // Class GaussPointItem 
+
+/** \brief InternalVariablesInterpolationProcess
+ * This utilitiy has as objective to interpolate the values inside elements (and conditions?) in a model part, using as input the original model part and the new one
+ * The process employs the projection.h from MeshingApplication, which works internally using a kd-tree 
+ */
+
+class InternalVariablesInterpolationProcess 
+    : public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+    
+    // General type definitions
+    typedef ModelPart::NodesContainerType                    NodesArrayType;
+    typedef ModelPart::ElementsContainerType              ElementsArrayType;
+    typedef ModelPart::ConditionsContainerType          ConditionsArrayType;
+    typedef Node<3>                                                NodeType;
+    typedef Geometry<NodeType>                                 GeometryType;
+    
+    // Type definitions for the tree
+    typedef GaussPointItem                                        PointType;
+    typedef PointType::Pointer                             PointTypePointer;
+    typedef std::vector<PointTypePointer>                       PointVector;
+    typedef PointVector::iterator                             PointIterator;
+    typedef std::vector<double>                              DistanceVector;
+    typedef DistanceVector::iterator                       DistanceIterator;
+    
+    // KDtree definitions
+    typedef Bucket< 3ul, PointType, PointVector, PointTypePointer, PointIterator, DistanceIterator > BucketType;
+    typedef Tree< KDTreePartition<BucketType> > KDTree;
+
+    /// Pointer definition of InternalVariablesInterpolationProcess
+    KRATOS_CLASS_POINTER_DEFINITION( InternalVariablesInterpolationProcess );
+      
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    // Class Constructor
+    
+    /**
+     * The constructor of the search utility uses the following inputs:
+     * @param rOriginMainModelPart: The model part from where interpolate values
+     * @param rDestinationMainModelPart: The model part where we want to interpolate the values
+     * @param ThisParameters: The parameters containing all the information needed
+     */
+    
+    InternalVariablesInterpolationProcess(
+        ModelPart& rOriginMainModelPart,
+        ModelPart& rDestinationMainModelPart,
+        Parameters ThisParameters = Parameters("{'allocation_size': 1000, 'bucket_size': 4, 'search_factor': 2, 'interpolation_type': 'LST', 'internal_variable_interpolation_list':[]}")
```

Comment:
this way of employing the Parameters prevents you from using the Validation mechanism (and the defaults).

take a look at how it is done for example in the test_parameters.py